tests/test_s3.cc

```diff
@@ -0,0 +1,94 @@
+#include <experimental/filesystem>
+
+#include "gtest/gtest.h"
+
+#include "h5s3/s3.h"
+#include "process.h"
+
+namespace fs = std::experimental::filesystem;
+namespace s3 = h5s3::s3;
+
+using process = h5s3::testing::process;
+
+class S3Test : public ::testing::Test {
+public:
+    static std::string ACCESS_KEY;
+    static std::string SECRET_KEY;
+    static std::unique_ptr<process> minio_proc;
+
+protected:
+
+    static void SetUpTestCase() {
+        auto path = fs::temp_directory_path();
+        std::vector<std::string> args = {
+            "testbin/minio",
+            "server",
+            "--config-dir", "testbin/minio-config/",
+            "--address", "localhost:9999",
+            "/tmp/minio/",
+        };
+        process::environment env = {
+            {"MINIO_ACCESS_KEY", ACCESS_KEY},
+            {"MINIO_SECRET_KEY", SECRET_KEY},
+            {"PATH", std::getenv("PATH")},
+        };
+        minio_proc = std::make_unique<process>(args, env);
```

Comment:
why is this heap allocated instead of just a value?

Reply:
talked about this in person.  we're using the unique pointer to allow the lifetime of the process to be shorter than the lifetime of the test suite.